lizrd/core/distributed.py

```diff
@@ -0,0 +1,59 @@
+from typing import Optional
+from functools import partial
+
+from torch.distributed.fsdp import MixedPrecision, CPUOffload
+from torch.nn.parallel import DistributedDataParallel as DDP
+import torch.nn as nn
+import torch
+
+from torch.distributed.fsdp import FullyShardedDataParallel as FSDP
+
+
+def custom_auto_wrap_policy(
+    module: nn.Module,
+    recurse: bool,
+    nonwrapped_numel: int,
+    # Additional custom arguments
+    min_num_params: int = int(1e8),
+) -> bool:
+    return nonwrapped_numel >= min_num_params
+
+
+def wrap_in_fsdp(
+    module: nn.Module,
+    rank: Optional[int],
+    param_precision: torch.dtype,
+    cast_inputs: bool,
+    mixed_precision_ignore_classes: list,
+    offload_params: bool,
+    print_model: bool,
+):
+    def _create_single_fsdp_module(module_to_wrap: nn.Module, precision: torch.dtype):
```

Comment:
please remove this function - it is no longer required to create a helper function here

Reply:
done